x-pack/plugins/security_solution/public/timelines/store/middlewares/helpers.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { State } from '../../../common/store/types';
+import { ALL_TIMELINE_QUERY_ID } from '../../containers/all';
+import type { inputsModel } from '../../../common/store/inputs';
+import { inputsSelectors } from '../../../common/store/inputs';
+
+/**
+ * Refreshes all timelines, so changes are propagated to everywhere on the page
+ */
+export function refreshTimelines(state: State) {
+  const allTimelineQuery = inputsSelectors.globalQueryByIdSelector()(state, ALL_TIMELINE_QUERY_ID);
+  if (allTimelineQuery.refetch != null) {
+    (allTimelineQuery.refetch as inputsModel.Refetch)();
```

Comment:
Is this `fn` really stored in redux? I guess this is what @kqualters-elastic was talking about. I'm not sure if we have a ticket yet for removing non-serializable state from redux. Do you mind creating a ticket for refactoring the store to remove fn references and classes? Thanks!

Reply:
sometimes it is, sometimes it's recreated from state/props in different places, it's quite a mess.